core/src/main/scala/kafka/admin/ConsumerGroupCommand.scala

```diff
@@ -126,7 +128,21 @@ object ConsumerGroupCommand extends Logging {
     private var consumer: KafkaConsumer[String, String] = _
 
     def listGroups(): List[String] = {
-      adminClient.listAllConsumerGroupsFlattened().map(_.groupId)
+      val result = adminClient.listConsumerGroups(
+        withTimeoutMs(new ListConsumerGroupsOptions()))
+
+      try {
+        val listings = result.all.get().asScala
+        listings.map(_.groupId).toList
+      }
+      catch {
+        case e : ExecutionException if e.getCause.isInstanceOf[KafkaException] =>
```

Comment:
I'm not sure how precisely we need to maintain the error reporting. I'm wondering if we can just raise `e.getCause` for all instances of `ExecutionException`.

Reply:
In this "special" case, it is not enough to raise `e.getCause`.

When there is a SASL authentication failure, ``listings.map(_.groupId).toList`` throws:
```
concurrent.ExecutionException
--> org.apache.kafka.common.KafkaException: Failed to find brokers to send ListGroups
    --> org.apache.kafka.common.errors.SaslAuthenticationException: Authentication failed due to invalid credentials with SASL mechanism SCRAM-SHA-256
```  

`KafkaAdminClient` throws an `ExecutionException` when we try to get all results of the returned futures of `listGroups` (called from `testConsumerGroupServiceWithAuthenticationFailure`). The cause of the ExecutionException is a `org.apache.kafka.common.KafkaException` "Failed to find brokers to send ListGroups". Its cause is the `SaslAuthenticationException` if there is an authentication failure.